### DISCUSSION TITLE
Please assist

### DISCUSSION THREAD

**Discussion by capssicum** [007revad/Synology_HDD_db#385] [General]
Kindly advise me if i have something wrong before i run the script , i have ds920 with 4 bay hdds in shr config , an ssd configured as ssd cashe for volume 1 and an extra ssd wich i am planing to use the script for in order to create volume 2 for docker conatiner manager
I read the script and conclued to put

/volume1/DS920/Synology_HDD_DB/syno_hdd_db.sh --showedits --noupdate --ram -S --autoupdate=3

As a schedule task at bootup
Can i now run the task and everything in my command is right and cope with my nas or is there something i need to change. Thanks

**Comment by 007revad** [comment]
For a scheduled task I'd add -e or --email

**Reply by capssicum** [reply]
/volume1/DS920/Synology_HDD_DB/syno_hdd_db.sh --showedits --noupdate --ram --email -S --autoupdate=3
I added email as adviced👍🏻
now it is going to read from my ssd not from shr hhd 
And create volume from storage pool , right revad ?

**Reply by 007revad** [reply]
DSM will read from your SSD if the SSD is a 2.5 inch SATA SSD. DSM does not get installed on NVMe drives (or drives in expansion units).

The script won't create the NVMe storage pool or volume. It just makes storage manager think your NVMe drives are Synology NVMe drives so you can create the NVMe storage pool and volume in storage manager.

**Reply by capssicum** [reply]
Then i should remove the -S from the task right?

**Reply by 007revad** [reply]
Yes

**Reply by capssicum** [reply]
Revad thank you very much for replying to me , thanks revad ❤️

**Comment by capssicum** [comment]
Dear david i ran the script but i cant still use the nvme to create the storage pool 
Here r the logs 

Task Scheduler has completed a triggered task.

Task: Syno HDD
Start time: Sun, 17 Nov 2024 00:55:22 GMT
Stop time: Sun, 17 Nov 2024 00:55:26 GMT
Current status: 0 (Normal)
Standard output/error:

Synology_HDD_db v3.5.103
DS920+ DSM 7.2.2-72806-1
StorageManager 1.0.0-00502

ds920+_host_v7 version 8057

Using options: --showedits --noupdate --ram --email --autoupdate=3
Running from: /volume1/DS920/Synology_HDD_DB/syno_hdd_db.sh

HDD/SSD models found: 2
ST18000NE000-2YY101,EN01,18000 GB
WD80EFZZ-68BTXN0,81.00A81,8001 GB

M.2 drive models found: 2
Lexar 512GB SSD,V1.16.2,512 GB
WDC WDS100T2B0C-00PXH0,233010WD,1000 GB

No M.2 PCIe cards found

No Expansion Units found

ST18000NE000-2YY101 already exists in ds920+_host_v7.db
WD80EFZZ-68BTXN0 already exists in ds920+_host_v7.db
Lexar 512GB SSD already exists in ds920+_host_v7.db
WDC WDS100T2B0C-00PXH0 already exists in ds920+_host_v7.db

Support disk compatibility already enabled.

Support memory compatibility already disabled.

Max memory already set to 20 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already disabled.

ST18000NE000-2YY101:
{
"EN01": {
"size_gb": 18000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 18000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

WD80EFZZ-68BTXN0:
{
"81.00A81": {
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 8000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

Lexar 512GB SSD:
{
"V1.16.2": {
"size_gb": 512,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 512,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

WDC WDS100T2B0C-00PXH0:
{
"233010WD": {
"size_gb": 1000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 1000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes
![Uploading IMG_3171.jpeg…]()

**Reply by capssicum** [reply]
![IMG_3172](https://github.com/user-attachments/assets/d6bec706-b279-4fe6-a908-ed9995ef1ca8)

**Comment by 007revad** [comment]
Have you rebooted?

**Comment by capssicum** [comment]
4 times and the task 4 times

**Reply by 007revad** [reply]
This is strange. I have no problem creating an NVMe volume on my DS720+ after running syno_hdd_db

Can you try https://github.com/007revad/Synology_enable_M2_volume

**Reply by capssicum** [reply]
May be it is not supported with this dsm version ?
![IMG_3178](https://github.com/user-attachments/assets/2b835ea2-2d08-41b2-9801-b18c8fa1ecd4)

**Reply by capssicum** [reply]
I tried the enable m2 script and still cant create and rebooted 2 times 
Task Scheduler has completed a triggered task.

Task: Enable m2 volume
Start time: Sun, 17 Nov 2024 17:13:31 GMT
Stop time: Sun, 17 Nov 2024 17:13:36 GMT
Current status: 0 (Normal)
Standard output/error:

Synology_HDD_db v3.5.103
DS920+ DSM 7.2.2-72806-1
StorageManager 1.0.0-00502

ds920+_host_v7 version 8057

Using options: --showedits --noupdate --ram --email --autoupdate=3
Running from: /volume1/DS920/Synology_HDD_DB/syno_hdd_db.sh

HDD/SSD models found: 2
ST18000NE000-2YY101,EN01,18000 GB
WD80EFZZ-68BTXN0,81.00A81,8001 GB

M.2 drive models found: 2
Lexar 512GB SSD,V1.16.2,512 GB
WDC WDS100T2B0C-00PXH0,233010WD,1000 GB

No M.2 PCIe cards found

No Expansion Units found

ST18000NE000-2YY101 already exists in ds920+_host_v7.db
WD80EFZZ-68BTXN0 already exists in ds920+_host_v7.db
Lexar 512GB SSD already exists in ds920+_host_v7.db
WDC WDS100T2B0C-00PXH0 already exists in ds920+_host_v7.db

Support disk compatibility already enabled.

Support memory compatibility already disabled.

Max memory already set to 20 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already disabled.

ST18000NE000-2YY101:
{
"EN01": {
"size_gb": 18000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 18000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

WD80EFZZ-68BTXN0:
{
"81.00A81": {
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 8000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

Lexar 512GB SSD:
{
"V1.16.2": {
"size_gb": 512,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 512,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

WDC WDS100T2B0C-00PXH0:
{
"233010WD": {
"size_gb": 1000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
},
"default": {
"size_gb": 1000,
"compatibility_interval": [
{
"compatibility": "support",
"not_yet_rolling_status": "support",
"fw_dsm_update_status_notify": false,
"barebone_installable": true,
"barebone_installable_v2": "auto",
"smart_test_ignore": false,
"smart_attr_ignore": false
}
]
}
}

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.

Synology_enable_M2_volume v1.1.22
DS920+ DSM 7.2.2-72806-1

StorageManager 1.0.0-00502

Using options: --email --autoupdate=3
Running from: /volume1/DS920/Synology_enable_m2_volume/syno_enable_m2_volume.sh
Backup successful.

Checking file.
202378 = 3168a
0003168a: 803E 00B8 0100 0000 7524 488B

Editing file.

Checking if file was successfully edited.
202378 = 3168a
0003168a: 803E 00B8 0100 0000 9090 488B
File successfully edited.

You can now create your M.2 storage pool in Storage Manager.

M.2 volume support already enabled.

The Synology needs to restart.
Type yes to reboot now.
Type anything else to quit (if you will restart it yourself).


From DS920

**Comment by capssicum** [comment]
David i used the enable m2 script and it worked , but containers like grafana cant see the storage pool 1 with an error : Partition details not found filesystem=/dev/sda1 
i used hdd scricpt too at boot up , any help ?

**Reply by 007revad** [reply]
A DS920+ does not use /dev/sda, /dev/sdb, dev/sdc etc. It uses /dev/sata1, /dev/stata2, /dev/sata3 etc.

So in grafana replace `/dev/sda1` with `/dev/sata1p1`
